plugin_tests/utils_test.py

```diff
@@ -162,9 +162,31 @@ def test_get_dataone_package_url(self):
         url = get_dataone_package_url(DataONELocations.prod_cn, pid)
         self.assertEqual(url, 'https://search.dataone.org/#view/'+pid)
 
+    def test_extract_orcid_id(self):
+        from server.utils import parse_jwt
+        from server.utils import extract_orcid_id
+
+        jwt = "eyJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJodHRwOlwvXC9vcmNpZC5vcm" \
```

Comment:
Is this a real DataONE JWT?

Reply:
I invalidated this JWT after using it.

Reply:
How about using PyJWT instead of manually parsing jwt tokens?

```
>>> import jwt
>>> print(jwt.decode(token, verify=False))
{'sub': 'http://orcid.org/0000-0002-1756-2128', 'fullName': 'Thomas Thelen', 'issuedAt': '2018-08-02T22:06:06.403+00:00', 'consumerKey': 'theconsumerkey', 'exp': 1533312366, 'userId': 'http://orcid.org/0000-0002-1756-2128', 'ttl': 64800, 'iat': 1533247566}
```

Does DataONE expose public RSA key via API? We could use that to verify the issuer.

Reply:
+1 on using a package instead of hand-coding this.

DataONE doesn't expose a public key no.